File: packages/sdk-dart/test/sdk_smoke_test.dart
import 'package:test/test.dart';
import 'package:paychain_sdk/paychain_sdk.dart';

void main() {
  group('PayChain Dart SDK smoke', () {
    test('exports health model and API surface', () {
      final health = Health((b) => b..status = 'ok');
      final sdk = PaychainSdk(basePathOverride: 'https://api.paychain.example');

      expect(health.status, 'ok');
      expect(sdk.getHealthApi(), isA<HealthApi>());
      expect(
        standardSerializers.serializeWith(Health.serializer, health),
        isNotNull,
      );
    });
  });
}

File: packages/sdk-php/tests/Smoke/SdkSmokeTest.php
<?php

namespace PayChainSdk\Test\Smoke;

use PayChainSdk\Api\HealthApi;
use PayChainSdk\Configuration;
use PayChainSdk\Model\Health;
use PHPUnit\Framework\TestCase;

final class SdkSmokeTest extends TestCase
{
    public function testHealthModelStoresStatus(): void
    {
        $health = new Health(['status' => 'ok']);

        self::assertSame('ok', $health->getStatus());
        self::assertTrue($health->valid());
    }

    public function testHealthApiUsesConfiguredHost(): void
    {
        $config = (new Configuration())->setHost('https://api.paychain.example');
        $api = new HealthApi(null, $config);

        self::assertSame('https://api.paychain.example', $api->getConfig()->getHost());
    }
}
